File: packages/sdks/python/tester-local/tests/test_start_consolidated_query.py
import os
import time
from dotenv import load_dotenv
import json
from generated.client import Metriport

load_dotenv()

api_key = os.environ.get("API_KEY")
patient_id = os.environ.get("PATIENT_ID")
base_url = os.environ.get("BASE_URL")

def test_start_consolidated_query():
    client = Metriport(api_key=api_key, base_url=base_url)

    print("Calling get_consolidated_query_status...")
    query_status = client.medical.fhir.get_consolidated_query_status(id=patient_id)
    print(f"queryStatus: {query_status}")

    print("Calling start_consolidated_query...")
    response = client.medical.fhir.start_consolidated_query(
        id=patient_id,
        resources="DocumentReference,Appointment",
        date_from="2021-03-01",
        date_to="2023-04-23"
    )
    print(f"response: {json.dumps(response.dict(), indent=2)}")

    print("Now, calling get_consolidated_query_status...")
    query_status = client.medical.fhir.get_consolidated_query_status(id=patient_id)
    print(f"queryStatus: {json.dumps(query_status.dict(), indent=2)}")

    print("Sleeping...")
    time.sleep(5)

    print("Calling get_consolidated_query_status again...")
    query_status = client.medical.fhir.get_consolidated_query_status(id=patient_id)
    print(f"queryStatus: {json.dumps(query_status.dict(), indent=2)}")

    print("Counting...")
    count = client.medical.fhir.count_patient_data(
        id=patient_id,
    )
    print(f"count: {json.dumps(count.dict(), indent=2)}")

File: packages/sdks/java/tester-local/src/test/java/com.metriport/TestStartConsolidatedQuery.java
package com.metriport;

import com.metriport.generated.Metriport;
import com.metriport.generated.core.RequestOptions;
import com.metriport.generated.resources.medical.fhir.types.ConsolidatedBundleUpload;
import com.metriport.generated.resources.medical.fhir.requests.GetPatientConsolidatedData;
import com.fasterxml.jackson.core.JsonProcessingException;


import io.github.cdimascio.dotenv.Dotenv;
import org.junit.jupiter.api.Test;
import com.metriport.generated.core.ApiError;

import java.util.concurrent.TimeUnit;

public class TestStartConsolidatedQuery {
    @Test
    public void startConsolidatedQuery() {
        Dotenv dotenv = Dotenv.load();

        String apiKey = dotenv.get("API_KEY");
        String patientId = dotenv.get("PATIENT_ID");
        String baseUrl = dotenv.get("BASE_URL");

        Metriport client = Metriport.builder()
                .apiKey(apiKey)
                .url(baseUrl)
                .build();

        System.out.println("Calling start_consolidated_query...");

        try {
            // Initialize RequestOptions

            GetPatientConsolidatedData request = GetPatientConsolidatedData.builder()
                .resources("Encounter")
                .dateFrom("2022-01-01")
                .dateTo("2022-12-31")
                .conversionType("pdf")
                .build();

            // Now, pass requestOptions to the method call
            var response = client.medical().fhir().startConsolidatedQuery(patientId, request);
            System.out.println("response: " + response);
        } catch (ApiError error) {
            System.err.println("API error: " + error.getMessage());
            error.printStackTrace();
        } catch (RuntimeException e) {
            System.err.println("Runtime exception: " + e.getMessage());
            e.printStackTrace();
        } catch (Exception e) {
            System.err.println("General exception: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
